### ISSUE TITLE
Mapping arrays with single entry to objects

### ISSUE THREAD

**Issue by bronek89** [CuyZ/Valinor#269]
Hi:) 

For start - thank you for creating and maintaining this library. I'm using it to map jsons inputs to objects from version `0.3`:) 

I have a problem with change introduced in this commit: https://github.com/CuyZ/Valinor/commit/72cba320f582c7cda63865880a1cbf7ea292d2b1

After change in handling objects with single property/argument
it is now harder to map some structures to objects.

For example - how to map below structure to classes?

```php
[
  'id' => 'xx-yy-zz',
  'info' => ['bio' => 'my description'],
]
```

It is simplified result from a real JSON api (that cannot be changed easily).

Up until now, I have something like this:

```php
final class UserId
{
    public string $value;
}

final class UserInfo
{
    public string $bio;
}

final class User
{
    public UserId $id;
    public UserInfo $info;
}
```

But on 1.0 it produces given error:

``` An error occurred at path info: Value array{bio: 'my description'} does not match type `string`.```

I've tried to introduce separate `Bio` class:

```php
final class Bio
{
    public string $value;
}

final class UserInfo
{
    public Bio $bio;
}
```

But it results in another error:

```An error occurred at path info: Value array{bio: 'my description'} does not match type `?`. ```

Both examples works good on version `0.17`

Maybe we can have an option to choose if mapper should create object with single property/argument from array?

**Comment by romm** [comment]
I've had multiple reports telling that the change introduced in 72cba320f582c7cda63865880a1cbf7ea292d2b1 brings difficulties to using the mapper; after giving it some thought I'm considering doing a revert of this change.

I'll keep you updated here, thank you for your patience.

**Comment by lookyman** [comment]
I just ran into this 😒 One quick and dirty solution I've found is to introduce a second, fake, nullable property with default value. From the example above:

```php
final class UserInfo
{
    public string $bio;
    public ?bool $fake = null;
}
```

Although I'm hoping for a revert anyway.

**Comment by magnetik** [comment]
I was bitten by this too. I have some use case where I map response from APIs where first level key must not be ignored.

**Comment by romm** [comment]
Hi guys, sorry for the inconvenience, a fix has been merged with #318; I plan to do a release ASAP so you can use it in your applications.

Thank you for your patience! 🙂

**Comment by romm** [comment]
Hi there, release was just published: https://github.com/CuyZ/Valinor/releases/tag/1.2.0

Enjoy!
